src/world/area_mgm/mgm_00/E0F980.c

```diff
@@ -1,9 +1,139 @@
 #include "mgm_00.h"
 
-INCLUDE_ASM(s32, "world/area_mgm/mgm_00/E0F980", func_802411E0_E0F980);
+void snd_start_sound(s32 soundID, u8 volume, u8 pan);
+void msg_draw_frame(s32 posX, s32 posY, s32 sizeX, s32 sizeY, s32 style, s32 palette, s32 fading, s32 bgAlpha, s32 frameAlpha);
 
-INCLUDE_ASM(s32, "world/area_mgm/mgm_00/E0F980", func_80241540_E0FCE0);
+extern Gfx D_80243C50_E123F0[];
 
-INCLUDE_ASM(s32, "world/area_mgm/mgm_00/E0F980", func_802416C0_E0FE60);
+#define RECORD_DISPLAY_MAP_VAR  0xA
+#define MINIGAME_TYPE_SMASH     0
+#define MINIGAME_TYPE_JUMP      1
 
-INCLUDE_ASM(s32, "world/area_mgm/mgm_00/E0F980", func_802416F8_E0FE98);
+#define FADE_IN_TIME  5
+#define FADE_OUT_TIME 5
+
+typedef enum RecordDisplayState {
+    RECORD_START_SHOW       = 0,
+    RECORD_STATE_FADE_IN    = 1,
+    RECORD_STATE_VISIBLE    = 2,
+    RECORD_START_HIDE       = 3,
+    RECORD_STATE_FADE_OUT   = 4,
+    RECORD_STATE_DONE       = 5
+} RecordDisplayState;
+
+typedef struct RecordDisplayData {
+/* 0x00 */ RecordDisplayState state;
+/* 0x04 */ s32 stateTimer;
+/* 0x08 */ s32 gameType;
+/* 0x0C */ s32 alpha;
+/* 0x10 */ s32 curAlpha;
+/* 0x14 */ s32 workerID;
+} RecordDisplayData; /* size = 0x18 */
+
+void N(draw_record_display)(RecordDisplayData* data, s32 alpha) {
+    if (alpha > 0) {
+        gSPDisplayList(gMasterGfxPos++, D_80243C50_E123F0);
+        gDPPipeSync(gMasterGfxPos++);
+        gDPSetPrimColor(gMasterGfxPos++, 0, 0, 16, 120, 24, alpha * 0.65);
+        gDPFillRectangle(gMasterGfxPos++, 48, 53, 272, 129);
+        gDPPipeSync(gMasterGfxPos++);
+        msg_draw_frame(45, 50, 230, 82, 6, 0, 1, (s32)(alpha * 0.55), alpha); // cast needed if signature isn't present
+        if (data->gameType == MINIGAME_TYPE_JUMP) {
+            draw_msg(MESSAGE_ID(0x8,0x1C), 66, 57, alpha, MSG_PAL_TEAL, 0);
+            draw_number(gPlayerData.jumpGamePlays,   219, 78,  TRUE, MSG_PAL_WHITE, alpha, 3);
+            draw_number(gPlayerData.jumpGameTotal,   219, 93,  TRUE, MSG_PAL_WHITE, alpha, 3);
+            draw_number(gPlayerData.jumpGameRecord,  219, 108, TRUE, MSG_PAL_WHITE, alpha, 3);
+        } else {
+            draw_msg(MESSAGE_ID(0x8,0x1D), 64, 57, alpha, MSG_PAL_TEAL, 0);
+            draw_number(gPlayerData.smashGamePlays,  219, 78,  TRUE, MSG_PAL_WHITE, alpha, 3);
+            draw_number(gPlayerData.smashGameTotal,  219, 93,  TRUE, MSG_PAL_WHITE, alpha, 3);
+            draw_number(gPlayerData.smashGameRecord, 219, 108, TRUE, MSG_PAL_WHITE, alpha, 3);
+        }
+
+        draw_msg(MESSAGE_ID(0x8,0x1E), 58,  78,  alpha, MSG_PAL_WHITE, DRAW_MSG_STYLE_MENU);
```

Comment:
space after comma here I think would be good for consistency - was this intentional, to save space?

Reply:
im regarding these as a single token atm which will be replaced in the future with either enum values (MSG_Minigame_JumpRecord) or a similar macro of the form MSG(Minigame, JumpRecord) which resolves to the former.

Reply:
We have named message ID macros already, see msg.yaml